mitodl/mitxonline
# 811: need financial assistance link on the checkout page

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/811

#### What's this PR do?
If a user adds a Product that offers Flexible Pricing, but the user does not currently have a Flexible Price, to their Cart then the user should be presented with a link that takes them to the Flexible Pricing Form for the Product in their Cart. 

Things to note: The  'Need financial assistance?' link is only hidden when the user has an approved flexible pricing request.

#### How should this be manually tested?

1. Login as a user and add a Course which the user does not have any approved Flexible Pricing to your Cart.
2. View the Cart and ensure that you can see the 'Need financial assistance?' link.
3. Click the link and ensure that you are navigated to the Flexible Pricing form associated with the Product in your Cart.
4. Submit a Flexible Pricing request and approve the request.
5. Visit the Cart again and ensure that you no longer see the 'Need financial assistance?' link.

#### Screenshots (if appropriate)
![Screen Shot 2022-08-16 at 9 06 12](https://user-images.githubusercontent.com/8311573/184888084-ae1b0654-de10-4466-a969-3a1ee70e49ac.png)
![Screen Shot 2022-08-16 at 9 06 03](https://user-images.githubusercontent.com/8311573/184888088-632e6e34-c69f-44a2-a4e2-4916d5e3c9a9.png)
![Screen Shot 2022-08-16 at 9 05 43](https://user-images.githubusercontent.com/8311573/184888090-1a3cc74b-0e2a-4afc-baff-8e700a793a40.png)
